### DISCUSSION TITLE
External Scaler KEDA v2: scaledObject.ScalerMetadata - Metadata Map is empty

### DISCUSSION THREAD

**Discussion by spalaka** [kedacore/keda#1255] [Q&A / Need Help] [answered]
Hi Experts,

**Need help, I couldn't read the Metadata in IsActive method**

I have developed the external scaler based on KEDA v2 (https://keda.sh/docs/2.0/concepts/external-scalers/)

I see all the connectivity is good and the scaler is triggering the methods IsActive, GetMetricSpec, GetMetrics, this has been observed through logs.

To progress on the logic based on the metadata configured in ScaledObject, I am not seeing metadata is getting in IsActive through scaledObject.ScalerMetadata function.

IsActive Code:
func (e *ExternalScaler) IsActive(ctx context.Context, scaledObject *pb.ScaledObjectRef) (*pb.IsActiveResponse, error) {
	serverURL := scaledObject.ScalerMetadata["server"]
	log.Printf("CHK - IsActive")
	log.Printf("CHK - IsActive Metadata serverURL", serverURL)
	log.Printf("CHK - IsActive ScaledObject Name ", scaledObject.Name)
	log.Printf("CHK - IsActive ScaledObject Namespace ", scaledObject.Namespace)
	log.Printf("CHK - IsActive ScaledObject ScalerMetadata ", scaledObject.ScalerMetadata)

	return &pb.IsActiveResponse{
		Result: true,
	}, nil
}

ScaledObject Configuration:
spec:
    pollingInterval: 10   # Optional. Default: 30 seconds
    cooldownPeriod: 100  # Optional. Default: 300 seconds
    minReplicaCount: 1   # Optional. Default: 0
    maxReplicaCount: 30  # Optional. Default: 100
    scaleTargetRef:
        apiVersion:    apps/v1
        kind:          StatefulSet
        name:          my-release144-pubsubplus
    triggers:
      - type: external
        metadata:
          scalerAddress: 110.150.1.102:6000
          server: "http://110.150.1.101:9000"
          vpnName: "eCom_VPN"
          backlog: "10"
          solaceObjectType: "Queue"
          solaceObject: "TestQ"
          configMode: "monitor"

Logs from Scaler:
2020/10/14 14:52:08 CHK - IsActive Metadata serverURL%!(EXTRA string=)
2020/10/14 14:52:08 CHK - IsActive ScaledObject Name %!(EXTRA string=scaledobject-v2-144)
2020/10/14 14:52:08 CHK - IsActive ScaledObject Namespace %!(EXTRA string=default)
2020/10/14 14:52:08 CHK - IsActive ScaledObject ScalerMetadata %!(EXTRA map[string]string=map[])
2020/10/14 14:52:08 CHK - GetMetricSpec
2020/10/14 14:52:11 CHK - GetMetricSpec
2020/10/14 14:52:11 CHK - GetMetrics
2020/10/14 14:52:18 CHK - IsActive
2020/10/14 14:52:18 CHK - IsActive Metadata serverURL%!(EXTRA string=)
2020/10/14 14:52:18 CHK - IsActive ScaledObject Name %!(EXTRA string=scaledobject-v2-144)
2020/10/14 14:52:18 CHK - IsActive ScaledObject Namespace %!(EXTRA string=default)
2020/10/14 14:52:18 CHK - IsActive ScaledObject ScalerMetadata %!(EXTRA map[string]string=map[])
2020/10/14 14:52:18 CHK - GetMetricSpec
2020/10/14 14:52:26 CHK - GetMetricSpec
2020/10/14 14:52:26 CHK - GetMetrics

Logs from keda-operator-metrics-apiserver:
1014 15:06:39.588841       1 clientconn.go:106] parsed scheme: ""
I1014 15:06:39.588888       1 clientconn.go:106] scheme "" not registered, fallback to default scheme
I1014 15:06:39.588922       1 passthrough.go:48] ccResolverWrapper: sending update to cc: {[{110.150.1.102:6000  <nil> 0 <nil>}] <nil> <nil>}
I1014 15:06:39.588940       1 clientconn.go:933] ClientConn switching balancer to "pick_first"
I1014 15:06:39.589017       1 clientconn.go:882] blockingPicker: the picked transport is not ready, loop back to repick
I1014 15:06:54.623467       1 clientconn.go:106] parsed scheme: ""
I1014 15:06:54.623526       1 clientconn.go:106] scheme "" not registered, fallback to default scheme
I1014 15:06:54.623582       1 passthrough.go:48] ccResolverWrapper: sending update to cc: {[{110.150.1.102:6000  <nil> 0 <nil>}] <nil> <nil>}
I1014 15:06:54.623627       1 clientconn.go:933] ClientConn switching balancer to "pick_first"
I1014 15:06:54.623753       1 clientconn.go:882] blockingPicker: the picked transport is not ready, loop back to repick
I1014 15:06:54.625489       1 clientconn.go:106] parsed scheme: ""
I1014 15:06:54.625535       1 clientconn.go:106] scheme "" not registered, fallback to default scheme
I1014 15:06:54.625629       1 passthrough.go:48] ccResolverWrapper: sending update to cc: {[{110.150.1.102:6000  <nil> 0 <nil>}] <nil> <nil>}
I1014 15:06:54.625650       1 clientconn.go:933] ClientConn switching balancer to "pick_first"

Thanks,
SKP

**Comment by zroubalik** [comment, marked as answer]
@ahmelsayed PTAL

**Reply by spalaka** [reply]
Hi @ahmelsayed & @zroubalik, any suggestions to fix this?

**Reply by spalaka** [reply]
Hi @ahmelsayed & @zroubalik, Can I consider this as a bug and wait for your response?

**Reply by zroubalik** [reply]
@spalaka sorry for the delay. If you still have the problem, could you please open an issue and close this? Thanks
